GVFS/GVFS.Platform.POSIX/NetCoreMethods.cs

```diff
@@ -1,101 +1,101 @@
-        /// <param name="psi">The ProcessStartInfo.</param>
-        /// <returns>The envp array.</returns>
-        public static string[] CreateEnvp(ProcessStartInfo psi)
-        {
-            var envp = new string[psi.Environment.Count];
-            int index = 0;
-            foreach (var pair in psi.Environment)
-            {
-                envp[index++] = pair.Key + "=" + pair.Value;
-            }
-            return envp;
-        }
-
-        // https://github.com/dotnet/corefx/blob/4464cceace7615502a1003bc7488f206a47c82ee/src/Common/src/Interop/Unix/System.Native/Interop.ForkAndExecProcess.cs#L61
-        public static unsafe void AllocNullTerminatedArray(string[] arr, ref byte** arrPtr)
-        {
-            int arrLength = arr.Length + 1; // +1 is for null termination
-
-            // Allocate the unmanaged array to hold each string pointer.
-            // It needs to have an extra element to null terminate the array.
-            arrPtr = (byte**)Marshal.AllocHGlobal(sizeof(IntPtr) * arrLength);
-            Debug.Assert(arrPtr != null);
-
-            // Zero the memory so that if any of the individual string allocations fails,
-            // we can loop through the array to free any that succeeded.
-            // The last element will remain null.
-            for (int i = 0; i < arrLength; i++)
-            {
-                arrPtr[i] = null;
-            }
-
-            // Now copy each string to unmanaged memory referenced from the array.
-            // We need the data to be an unmanaged, null-terminated array of UTF8-encoded bytes.
-            for (int i = 0; i < arr.Length; i++)
-            {
-                byte[] byteArr = Encoding.UTF8.GetBytes(arr[i]);
-
-                arrPtr[i] = (byte*)Marshal.AllocHGlobal(byteArr.Length + 1); //+1 for null termination
-                Debug.Assert(arrPtr[i] != null);
-
-                Marshal.Copy(byteArr, 0, (IntPtr)arrPtr[i], byteArr.Length); // copy over the data from the managed byte array
-                arrPtr[i][byteArr.Length] = (byte)'\0'; // null terminate
-            }
-        }
-
-        // https://github.com/dotnet/corefx/blob/4464cceace7615502a1003bc7488f206a47c82ee/src/Common/src/Interop/Unix/System.Native/Interop.ForkAndExecProcess.cs#L92
-        public static unsafe void FreeArray(byte** arr, int length)
-        {
-            if (arr != null)
-            {
-                // Free each element of the array
-                for (int i = 0; i < length; i++)
-                {
-                    if (arr[i] != null)
-                    {
-                        Marshal.FreeHGlobal((IntPtr)arr[i]);
-                        arr[i] = null;
-                    }
-                }
-
-                // And then the array itself
-                Marshal.FreeHGlobal((IntPtr)arr);
-            }
-        }
-
-    }
-}
+// This file isn't generated, but this comment is necessary to exclude it from StyleCop analysis.
```

Comment:
This comment applies in a few places, but I see in some files there are a lot of whitespace changes.  Unfortunately the line endings in the repo are kind of messy right now.  Could you try to leave whatever line endings are already being used in a file as-is rather than updating them all?

cc: @jrbriggs for his thoughts as well.

Reply:
Good catch. I had whitespace diffs disabled.

Reply:
I can try to fix this while @kivikakk is out; it likely stems from my using LF on the Linux-exclusive files and then having those converted to the POSIX versions, instead of going from the Mac originals to the POSIX variant.

Reply:
OK ... as a pure whitespace fixup, I've added f6aac4a which puts the line endings back as they were in the Mac originals, but makes no other edits.